### ISSUE TITLE
TypeError: expected str, bytes or os.PathLike object, not Path

### ISSUE THREAD

**Issue by gnlenfn** [DataDog/dd-trace-py#9547]
### Summary of problem
when my app start up, each container shows error below
I found stack trace in `ddtrace-run -i`
```
Summary
Unable to build package file mapping, please report this to https://github.com/DataDog/dd-trace-py/issues
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/ddtrace/internal/utils/cache.py", line 124, in _
    retval, exc = f.__callonce_result__  # type: ignore[attr-defined]
AttributeError: 'function' object has no attribute '__callonce_result__'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/ddtrace/internal/packages.py", line 106, in _package_file_mapping
    _path = fspath(f.locate())
TypeError: expected str, bytes or os.PathLike object, not Path
```

### Which version of dd-trace-py are you using?
2.8.0

### Which version of pip are you using?
23.0.1

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
aiohttp==3.9.5
aiosignal==1.3.1
annotated-types==0.7.0
anyio==4.4.0
async-timeout==4.0.3
attrs==23.2.0
beautifulsoup4==4.12.3
bidict==0.23.1
bs4==0.0.2
bytecode==0.15.1
cattrs==23.2.3
certifi==2024.6.2
charset-normalizer==3.3.2
click==8.1.7
cmudict==1.0.23
Cython==3.0.10
dataclasses-json==0.6.7
ddsketch==3.0.1
ddtrace==2.8.0
Deprecated==1.2.14
Distance==0.1.3
distro==1.9.0
emoji==1.2.0
envier==0.5.1
environs==9.5.0
exceptiongroup==1.2.1
fastapi==0.110.0
frozenlist==1.4.1
GeoAlchemy2==0.15.1
greenlet==3.0.3
grpcio==1.63.0
h11==0.14.0
hangul-jamo==1.0.1
httpcore==1.0.5
httpx==0.27.0
idna==3.7
importlib_metadata==7.1.0
importlib_resources==6.4.0
iniconfig==2.0.0
jamo==0.4.1
jsonpatch==1.33
jsonpointer==3.0.0
kollocate==0.0.2
koparadigm==0.10.0
kss==6.0.4
langchain==0.1.9
langchain-community==0.0.24
langchain-core==0.1.27
langchain-openai==0.0.7
langchain-pinecone==0.0.3
langsmith==0.1.77
marshmallow==3.21.3
mecab-python===0.996-ko-0.9.2
milvus-lite==2.4.7
multidict==6.0.5
mypy-extensions==1.0.0
networkx==3.3
numpy==1.26.4
openai==1.13.3
opentelemetry-api==1.25.0
orjson==3.10.4
packaging==23.2
pandas==2.2.2
pecab==1.0.8
pinecone-client==3.2.2
pluggy==1.5.0
protobuf==5.27.1
psycopg2-binary==2.9.6
pyarrow==16.1.0
pybind11==2.9.2
pydantic==2.7.3
pydantic_core==2.18.4
pymilvus==2.4.3
PyPika==0.48.9
pytest==8.2.2
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-json-logger==2.0.7
python-mecab-kor==1.2.8
pytz==2024.1
PyYAML==6.0
redis==5.0.1
regex==2024.5.15
requests==2.32.3
scipy==1.13.1
six==1.16.0
sniffio==1.3.1
soupsieve==2.5
SQLAlchemy==2.0.30
sqlparse==0.5.0
sse-starlette==2.1.0
starlette==0.36.3
tenacity==8.3.0
tiktoken==0.7.0
tomli==2.0.1
tossi==0.3.1
tqdm==4.66.4
typing-inspect==0.9.0
typing_extensions==4.12.2
tzdata==2024.1
ujson==5.10.0
Unidecode==1.3.8
urllib3==2.2.1
uvicorn==0.27.1
Whoosh==2.7.4
wikipedia==1.4.0
Wikipedia-API==0.6.0
wrapt==1.16.0
xlrd==1.2.0
xmltodict==0.13.0
yarl==1.9.4
zipp==3.19.2
</details>

### What is the result that you expected?
this error occurs in every containers. and my log is full of errors when it starts up.
It can be handled as multi line logs, but I don't want this error log shows up

**Comment by emmettbutler** [comment]
Hi @gnlenfn, and thanks for the bug report.

Looking at the diff, it seems fairly likely to me that this was fixed in #9047. Could you try updating to 2.10.0rc2 and see if that resolves the issue?

cc @P403n1x87

**Comment by gnlenfn** [comment]
@emmettbutler 
sorry for late reply!

I updated ddtrace version, but still got a same message.
ddtrace looks like works fine, but datadog shows error log.
